### ISSUE TITLE
Is it relevant to add a property to encode the moment where an ontology became deprecated ? 

### ISSUE THREAD

**Issue by hibadjebabria169** [agroportal/project-management#379]
**Question:** Knowing that an ontology can be marked as deprecated, is there a way to automate the edition of this property to encode the moment where an ontology became deprecated. Similarly, can we edit this property when a new submission is uploaded.

**Comment by jonquet** [comment]
This is the role of the "valid" date metadata property. This date shall more or less encode the moment where an ontology become deprecated.

**Comment by syphax-bouazzouni** [comment]
In fact, there is no need to have a `deprecated` attribute if it can be infrared from another one like a `date`. At least in the UI, the user does not need to fill in two properties(i.e deprecated and the date), he should only fill in the date and we will automatically infer/fill the deprecated attribute

**Comment by hibadjebabria169** [comment]
The property `deprecated` can be true even the ontology in the prodcution phase which means it's not retired and it's still valid yet

**Comment by jonquet** [comment]
Those are the defintions for the Validity date property: 
- DCTERMS: Date (often a range) of validity of a resource.
- PROV: Invalidation is the start of the destruction, cessation, or expiry of an existing entity by an activity. The entity is no longer available for use (or further invalidation) after invalidation. Any generation or usage of an entity precedes its invalidation.
- SCHEMA: The end date and time of the item (in ISO 8601 date format).

Let's assume that "deprecated=true" is the beginning o the "invalidation process" and status=retired is the end. 
It means that : 
- If deprecated=true, then valid="date of the day deprecated is set to true" if previously empty
- If status=retired => deprecated=true then valid="date of the day deprecated is set to true" if previously empty
- If valid < "date of the day" then deprecated =true
- If new submission then valid of old submission is set to date of modification of new submission or if empty date of uploading in the portal

**Comment by jonquet** [comment]
Complete rules in https://github.com/agroportal/project-management/issues/381 when discussion is over.

**Comment by syphax-bouazzouni** [comment]
Also a slight proposition, what do you think of **adding `deprecated` to status possible values** , instead of having a boolean attribute for just deprecated? (my goal is to get rid of the complexity here of interdependent properties as much as possible )

**Comment by jonquet** [comment]
Well one could rapidly say yes... but in fact no. 

Although it would be ok to have a new status "deprecated" but this will be less precise than the situation 
where deprecated=true and status=somethingelse
Still I could cope with this... 

BUT deprecated is actually part of the MOD 1.4 we do implement: https://github.com/sifrproject/MOD-Ontology/blob/master/versions/mod-v1.4_profile.rdfs#L381

In other words, in MOD we decided to keep both properties because one of them is super standard (owl:deprecated) and because "omv:status" can be used to refine...  but no one actually put on paper the dependencies btw properties!

**Comment by imadbourouche** [comment]
@jonquet what is the status of this issue ?

**Comment by jonquet** [comment]
We can close this discussion.
